activerecord/lib/active_record/connection_adapters/mysql/quoting.rb

```diff
@@ -93,7 +93,7 @@ def unquote_identifier(identifier)
         # of Strings since MySQL adapters are able to handle those classes more efficiently.
         def type_cast(value) # :nodoc:
           case value
-          when ActiveSupport::TimeWithZone
+          when ActiveSupport::TimeWithZone, Time
             # We need to check explicitly for ActiveSupport::TimeWithZone because
             # we need to transform it to Time objects but we don't want to
             # transform Time objects to themselves.
```

Comment:
Hum, I probably need to update that comment.

Reply:
This was added in https://github.com/rails/rails/commit/1ba87a80b6a1023c8f2a2183880d7a62c442413b, not 100% sure I understand all the consequences here.

Reply:
For `Time` we could skip the `getutc` / `getlocal` if it's already in the right one and save an allocation, whereas (per the existing comment) we _must_ always convert for TWZ.

Reply:
Makes sense.

Reply:
Done.